src/Sarif/Core/Run.cs

```diff
@@ -220,6 +220,14 @@ public bool ShouldSerializeGraphs()
             return this.Graphs.HasAtLeastOneNonDefaultValue(Graph.ValueComparer);
         }
 
+        public bool ShouldSerializeAutomationDetails()
+        {
+            return this.AutomationDetails?.Description != null ||
```

Comment:
>AutomationDetails?.Description [](http://example.com/codeflow?start=24&length=30)

Why should we serialize an empty description? suggest using is null or whitespace...

Reply:
The Description isn't a string, it is a Message type. It's why I didn't change to NullOrWhiteSpace check